src/box/iproto.cc

```diff
@@ -1720,6 +1808,79 @@ tx_inject_delay(void)
 	});
 }
 
+static void
+tx_process_begin(struct cmsg *m)
+{
+	struct iproto_msg *msg = tx_accept_msg(m);
+	struct obuf *out;
+	struct iproto_stream *stream = msg->stream;
+
+	if (tx_check_schema(msg->header.schema_version))
+		goto error;
+
+	if (box_txn_begin() != 0)
+		goto error;
+
+	stream->txn = in_txn();
+	txn_detach();
```

Comment:
One of the reasons why I asked to make `txn_detach` return the txn was that you could make this: `stream->txn = txn_detach();`.

Reply:
Fixed